agreement/persistence.go

```diff
@@ -63,6 +63,8 @@ func encode(t timers.Clock, rr rootRouter, p player, a []action, reflect bool) (
 		s.Player = protocol.Encode(&p)
 	}
 	s.Clock = t.Encode()
+	s.ActionTypes = make([]actionType, len(a))
+	s.Actions = make([][]byte, len(a))
```

Comment:
That's not quite right if we still `append`.  Either need to allocate 0 length and len(a) cap, or allocate len(a) but assign into it instead of append.

Reply:
good catch JJ!

Reply:
opted for the latter and thanks for catching it

Reply:
Also just added two actions into the existing test case that would have caught this issue.